docs/concept-workflows.md

```diff
@@ -60,70 +240,45 @@ public class SubscriptionWorkflowImpl implements SubscriptionWorkflow {
-- A single Workflow has a limited size. For example, it cannot execute 100k Activities. Child Workflows can be used to partition the problem into smaller chunks. One parent with 1000 children each executing 1000 Activities is 1 million executed Activities.
-- A child Workflow can be used to manage some resource using its Id to guarantee uniqueness. For example, a Workflow that manages host upgrades can have a child Workflow per host (host name being a Workflow Id) and use them to ensure that all operations on the host are serialized.
-- A child Workflow can be used to execute some periodic logic without blowing up the parent history size. When a parent starts a child, it executes periodic logic calling that continues as many times as needed, then completes. From the parent point of view, it is just a single child Workflow invocation.
+The idiomatic way to handle indefinitely running Workflows is to use the "Continue-as-new" feature, which is available in all SDKs.
+For example a reasonable cutoff point might be once a day for high volume Workflows.
 
-The main limitation of a child Workflow versus collocating all the application logic in a single Workflow is lack of the shared state. Parent and child can communicate only through asynchronous signals. But if there is a tight coupling between them, it might be simpler to use a single Workflow and just rely on a shared object state.
+The "Continue-as-new" feature completes the current Workflow execution and start a new execution with the same Workflow Id, passing it the appropriate parameters for it to continue.
+This keeps the event history within limits, but continues the logic execution.
 
-We recommended starting from a single Workflow implementation if your problem has bounded size in terms of number of executed Activities and processed signals. It is more straightforward than multiple asynchronously communicating Workflows.
+:::note
 
-## Workflow Timeouts
+If you are using [Signals](https://docs.temporal.io/docs/concept-signals/), you should make sure to do an asynchronous drain on the Signal channel or the Signals will be lost.
 
-It's often necessary to limit the amount of time a specific Workflow can be running. To support this, the following three parameters can be provided to Workflow options:
+:::
 
-- `WorkflowExecutionTimeout` maximum time a Workflow should be allowed to run including retries and continue as new. Use `WorkflowRunTimeout` to limit execution time of a single run.
-- `WorkflowRunTimeout` maximum time a single Workflow run should be allowed.
-- `WorkflowTaskTimeout` timeout for processing a Workflow task starting from the point when a worker pulled the task. If a decision task is lost, it is retried after this timeout.
+**How do I handle a Worker process failure/restart in my Workflow?**
 
-## Workflow Retries
+You do not. The Workflow code is completely oblivious to any Worker failures or downtime.
+It is even oblivious to the Temporal Server itself.
```

Comment:
> It is even oblivious to the Temporal Server itself.

I would remove this sentence. It is not factually correct as the service does react to timeouts caused by worker failures and is not something users care about.

Reply:
> recommend adding note on infinitely long running workflows: https://github.com/temporalio/documentation/pull/301/files#diff-0aa99244cc1ce21b5410cb97524ce95bf4ccee0834c8bcfe6fdd27ed40bbda12R86

Ya, it is at the bottom, part of the FAQ section - 
https://github.com/temporalio/documentation/pull/305/files#diff-0aa99244cc1ce21b5410cb97524ce95bf4ccee0834c8bcfe6fdd27ed40bbda12R252